os/tools/mkldscript.py

```diff
@@ -79,26 +79,42 @@
 
 ram_end = int(CONFIG_RAM_SIZE) + int(CONFIG_RAM_START, 16)
 
+if ram_end % 4096 != 0 :
+    print("RAM end should be 4KB aligned")
```

Comment:
Is the 4KB check is coupled with l2 page size?

Reply:
Yes. For apps we always allocate l2 pages.